diff --git a/vllm/v1/engine/utils.py b/vllm/v1/engine/utils.py
--- a/vllm/v1/engine/utils.py
+++ b/vllm/v1/engine/utils.py
@@ -180,8 +180,9 @@ def get_device_indices(device_control_env_var: str, local_dp_rank: int,
     this will select devices 2 and 3 for local_dp_rank=1.
     """
     try:
+        offset = int(os.getenv("START_DEVICE_ID", 0))
         value = ",".join(
-            str(current_platform.device_id_to_physical_device_id(i))
+            str(current_platform.device_id_to_physical_device_id(i+offset))
             for i in range(local_dp_rank * world_size, (local_dp_rank + 1) *
                            world_size))
     except IndexError as e:
