diff --git a/vllm/v1/request.py b/vllm/v1/request.py
--- a/vllm/v1/request.py
+++ b/vllm/v1/request.py
@@ -309,6 +309,7 @@ class RequestStatus(enum.IntEnum):
     FINISHED_IGNORED = enum.auto()
     FINISHED_ERROR = enum.auto()
     FINISHED_REPETITION = enum.auto()
+    FINISHED_MIGRATED = enum.auto()
 
     def __str__(self) -> str:
         return self.name
@@ -333,4 +334,5 @@ _FINISHED_REASON_MAP = {
     RequestStatus.FINISHED_ERROR: FinishReason.ERROR,
     RequestStatus.WAITING_FOR_STREAMING_REQ: FinishReason.STOP,
     RequestStatus.FINISHED_REPETITION: FinishReason.REPETITION,
+    RequestStatus.FINISHED_MIGRATED: FinishReason.MIGRATE,
 }
